src/core/Microsoft.PowerToys.Settings.UI/ViewModels/KeyboardManagerViewModel.cs

```diff
@@ -22,34 +27,110 @@ public class KeyboardManagerViewModel : Observable
         private const string EditShortcutActionName = "EditShortcut";
         private const string EditShortcutActionValue = "Open Edit Shortcut Window";
         private const string JsonFileType = ".json";
-        private const string ConfigFileMutexName = "PowerToys.KeyboardManager.ConfigMutex";
-        private const int ConfigFileMutexWaitTimeoutMiliSeconds = 1000;
+        private const string ProfileFileMutexName = "PowerToys.KeyboardManager.ConfigMutex";
+        private const int ProfileFileMutexWaitTimeoutMiliSeconds = 1000;
+
+        private readonly CoreDispatcher dispatcher;
+        private readonly FileSystemWatcher watcher;
 
         private ICommand remapKeyboardCommand;
         private ICommand editShortcutCommand;
-        private FileSystemWatcher watcher;
         private KeyboardManagerSettings settings;
-
-        public ICommand RemapKeyboardCommand => remapKeyboardCommand ?? (remapKeyboardCommand = new RelayCommand(OnRemapKeyboard));
-
-        public ICommand EditShortcutCommand => editShortcutCommand ?? (editShortcutCommand = new RelayCommand(OnEditShortcut));
+        private KeyboardManagerProfile profile;
+        private GeneralSettings generalSettings;
 
         public KeyboardManagerViewModel()
         {
+            dispatcher = Window.Current.Dispatcher;
             if (SettingsUtils.SettingsExists(PowerToyName))
             {
                 // Todo: Be more resillent while reading and saving settings.
                 settings = SettingsUtils.GetSettings<KeyboardManagerSettings>(PowerToyName);
+
+                // Load profile.
+                if (!LoadProfile())
+                {
+                    profile = new KeyboardManagerProfile();
+                }
             }
             else
             {
                 settings = new KeyboardManagerSettings(PowerToyName);
                 SettingsUtils.SaveSettings(settings.ToJsonString(), PowerToyName);
             }
 
-            watcher = Helper.GetFileWatcher(PowerToyName, settings.Properties.ActiveConfiguration.Value + JsonFileType, OnConfigFileUpdate);
+            if (SettingsUtils.SettingsExists())
+            {
+                generalSettings = SettingsUtils.GetSettings<GeneralSettings>(string.Empty);
+            }
+            else
+            {
+                generalSettings = new GeneralSettings();
+                SettingsUtils.SaveSettings(generalSettings.ToJsonString(), string.Empty);
+            }

```

Comment:
Why are we initializing the general settings here? Shouldn't be done in the view model for the general page? Or is that the new approach that all powertoys are using?

Reply:
You are right, we should load the `GeneralSettings` file only once in memory and share it between the different pages.